### PR TITLE
video: reset `display_sync_error` when resetting state

### PR DESCRIPTION
**Pull request by llyyr**
This would cause mpv to, in some very specific scenarios, have a negative vsync_offset after seeking which would result in mpv requesting a pts before the first frame to libplacebo.

Fix it by setting it to 0 when we reset state, such as after seeking.

Fixes: https://github.com/mpv-player/mpv/issues/12813

### CODE REVIEW

**Comment by llyyr** [comment]
We may be able to revert this commit now https://github.com/mpv-player/mpv/commit/640c07fb19b7ea11f1a97784e517c38449f816d0 but I'll have to test that to be sure

File: player/video.c [hunk 1]
```diff
@@ -119,6 +119,7 @@ void reset_video_state(struct MPContext *mpctx)
     mpctx->mistimed_frames_total = 0;
     mpctx->drop_message_shown = 0;
     mpctx->display_sync_drift_dir = 0;
+    mpctx->display_sync_error = 0.0;
```
Comment (kasper93) [inline player/video.c:122]:
```suggestion
    mpctx->display_sync_error = 0;
```

Comment (llyyr) [reply player/video.c:122]: I originally had it like that but changed it for symmetry with https://github.com/mpv-player/mpv/blob/master/player/loadfile.c#L1601 but I guess following local style makes more sense

Comment (kasper93) [reply player/video.c]: We generally do `0`, and even few lines about `last_av_difference` is double too.

Comment (Dudemanguy) [reply player/video.c]: I mean it is a double though. Not like the implicit cast is the end of the world I guess.

**Review by Dudemanguy — APPROVED** [review]
LGTM. There's several "flicker with interpolation while seeking" issues open right now. Does this fix some more of them?

**Comment by llyyr** [comment]
> There's several "flicker with interpolation while seeking" issues open right now. Does this fix some more of them?

This probably only fixes bugs reported as a result of https://github.com/haasn/libplacebo/commit/e91f76515cf0c10b40880eb8b9bc4e5287ec5d01

I tested a couple which still had samples available and they still flicker

**Pull request merged**